lib/fileset/default.nix

```
@@ -709,4 +710,31 @@ in {
       throw "lib.fileset.gitTrackedWith: Expected the second argument (${toString path}) to point to a local working tree of a Git repository, but it's not."
     else
       _mirrorStorePath path fetchResult.outPath;
+
+  /*
+    Create a file set from a path that may or may not exist:
+    - If the path does exist, the path is [coerced to a file set](#sec-fileset-path-coercion).
+    - If the path does not exist, a file set containing no files is returned.
+
+    Type:
+      maybeMissing :: Path -> FileSet
+
+    Example:
+      # All files in the current directory, but excluding main.o if it exists
+      difference ./. (maybeMissing ./main.o)
+  */
+  maybeMissing =
+    path:
+    if ! isPath path then
+      if isStringLike path then
+        throw ''
+          lib.fileset.maybeMissing: Argument ("${toString path}") is a string-like value, but it should be a path instead.''
```

```suggestion
          lib.fileset.maybeMissing: Argument ("${toString path}") is a string-like value, but it should be a path instead. See https://nixos.org/manual/nix/stable/language/values#type-path''
```

I don't think the word "path" is enough for novices.

---

Done in https://github.com/NixOS/nixpkgs/pull/265964/commits/701cff3d368cc0bd2f5ddc54fd658c5cb37e68d0!